akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -441,7 +441,7 @@ object EntityTypeKey {
  *
  * Not for user extension.
  */
-@DoNotInherit abstract class EntityRef[M] extends RecipientRef[M] {
+@DoNotInherit abstract class EntityRef[-M] extends RecipientRef[M] {
```

Comment:
I checked javap before and after and it doesn't seem to make a difference but maybe someone who knows more about scala/java compat may know differently.

This is to make the asJava method compile

```
public abstract class akka.cluster.sharding.typed.javadsl.EntityRef<M> implements akka.actor.typed.RecipientRef<M> {
  public abstract void tell(M);
  public abstract <Res> java.util.concurrent.CompletionStage<Res> ask(akka.japi.function.Function<akka.actor.typed.ActorRef<Res>, M>, java.time.Duration);
  public abstract <Res> java.util.concurrent.CompletionStage<Res> askWithStatus(scala.Function1<akka.actor.typed.ActorRef<akka.pattern.StatusReply<Res>>, M>, java.time.Duration);
  public akka.cluster.sharding.typed.scaladsl.EntityRef<M> asScala();
  public akka.cluster.sharding.typed.javadsl.EntityRef();
}


Compiled from "ClusterSharding.scala"
public abstract class akka.cluster.sharding.typed.javadsl.EntityRef<M> implements akka.actor.typed.RecipientRef<M> {
  public abstract void tell(M);
  public abstract <Res> java.util.concurrent.CompletionStage<Res> ask(akka.japi.function.Function<akka.actor.typed.ActorRef<Res>, M>, java.time.Duration);
  public abstract <Res> java.util.concurrent.CompletionStage<Res> askWithStatus(scala.Function1<akka.actor.typed.ActorRef<akka.pattern.StatusReply<Res>>, M>, java.time.Duration);
  public akka.cluster.sharding.typed.scaladsl.EntityRef<M> asScala();
  public akka.cluster.sharding.typed.javadsl.EntityRef();
}
```

Reply:
I can change the impl to a cast if we think this is a problem

Reply:
In Scala it is `EntityRef[-M]` and that makes it possible to:
```
    val counterOne: EntityRef[Counter.Command] = sharding.entityRefFor(TypeKey, "counter-1")
    val ref2: EntityRef[Counter.GetValue] = counterOne
```
and that was introduced in https://github.com/akka/akka/pull/27081, but only for scaladsl

In Java that doesn't matter. Corresponding not possible with or without the `-`.

I guess it should be alright to change.